### PR TITLE
Review only search implemented

### PR DESCRIPTION
**Pull request by adeshp96**
I have implemented the reviews only search for release groups for now. Tested too. Doesn't works slow on my PC. A checkbox is also implemented for this. If you check the checkbox for artist or event nothing changes. Whereas for release groups it will filter for results with reviews only. Try it once. I don't think it is too slow. Also, I think it's a crucial feature because, if someone searches on critiquebrainz they want critiques too show up too and not just tracks(in case of release groups), so the user can simply filter the ones with reviews. For those who think otherwise, the checkbox will help :D

### CODE REVIEW

**Comment by gentlecat** [comment]
Same comment about strings from another PR applies here as well. See https://github.com/metabrainz/critiquebrainz/pull/14#issuecomment-164618006.

File: critiquebrainz/frontend/templates/review/modify/base.html [hunk 1]
```diff
@@ -56,6 +56,7 @@
     {% block license_agreement_input %} {% endblock %}
   </form>
   {% block buttons %} {% endblock %}
+  <br /><small class="text-warning"><em>{{ _('Note: Drafts are not private!') }}</em></small>
```
Comment (gentlecat) [inline critiquebrainz/frontend/templates/review/modify/base.html]: Is this from another pull request? You should keep each in separate branches based on *master*.

Comment (adeshp96) [inline critiquebrainz/frontend/templates/review/modify/base.html]: yeah i realized this problem a bit late :/  so should I submit another pull request? I can do this for the other pull requests too, perhaps making everything look a lot more cleaner? Either way, I will note this in future. Thank you

**Comment by adeshp96** [comment]
I have implemented this feature for search for any search query. Please let me know if there is any problem.

**Comment by adeshp96** [comment]
Merge conflict resolved.

File: critiquebrainz/frontend/views/search.py [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 from critiquebrainz.frontend.external import musicbrainz
 
 search_bp = Blueprint('search', __name__)
 
 RESULTS_LIMIT = 10
 
 
-def search_wrapper(query, type, offset=None):
-    if query:
+def search_wrapper(query, type, offset=None,review_only=False):
```
Comment (gentlecat) [inline critiquebrainz/frontend/views/search.py]: There should be a space after `,`.

File: critiquebrainz/frontend/views/search.py [hunk 3]
```diff
@@ -16,17 +18,49 @@ def search_wrapper(query, type, offset=None):
...
+        elif type == "event":
+            count, results = musicbrainz.search_events(query)
+        elif type == "release-group":
+            count, results = musicbrainz.search_release_groups(query)
+        else:
+            count, results = 0, []
     else:
         count, results = 0, []
+    if review_only is True :
+        if type== "artist":
```
Comment (gentlecat) [inline critiquebrainz/frontend/views/search.py]: Space before `==`.

File: critiquebrainz/frontend/views/search.py [hunk 4]
```diff
@@ -16,17 +18,49 @@ def search_wrapper(query, type, offset=None):
...
+            count, results = musicbrainz.search_artists(query)
+        elif type == "event":
+            count, results = musicbrainz.search_events(query)
+        elif type == "release-group":
+            count, results = musicbrainz.search_release_groups(query)
+        else:
+            count, results = 0, []
     else:
         count, results = 0, []
+    if review_only is True :
```
Comment (gentlecat) [inline critiquebrainz/frontend/views/search.py]: No space before `:`.

File: critiquebrainz/frontend/views/search.py [hunk 5]
```diff
@@ -16,17 +18,49 @@ def search_wrapper(query, type, offset=None):
...
+        if type== "artist":
+             fresults=[]
```
Comment (gentlecat) [inline critiquebrainz/frontend/views/search.py]: Spaces around `=`.

File: critiquebrainz/frontend/views/search.py [hunk 6]
```diff
@@ -16,17 +18,49 @@ def search_wrapper(query, type, offset=None):
...
+             for group in results:
+                 count, release_groups = musicbrainz.browse_release_groups(artist_id=group['id'], release_types=["album"])
+                 for release in release_groups:
+                     if(Review.list(entity_id=release['id'])[0]):
```
Comment (gentlecat) [inline critiquebrainz/frontend/views/search.py]: You don't need to use parentheses with `if` statement.

**Comment by gentlecat** [comment]
Please clean up your code.

**Comment by gentlecat** [comment]
ok to test

File: critiquebrainz/frontend/views/search.py [hunk 7]
```diff
@@ -18,17 +19,50 @@ def search_wrapper(query, type, offset=None):
...
 
 
 @search_bp.route('/')
 def index():
     query = request.args.get('query')
     type = request.args.get('type')
-    count, results = search_wrapper(query, type)
-    return render_template('search/index.html', query=query, type=type, results=results, count=count, limit=RESULTS_LIMIT)
+    if(request.args.get('review-only')=="on"):
+        review_only=True;
```
Comment (gentlecat) [inline critiquebrainz/frontend/views/search.py]: You don't need to put a semicolon after statements in Python.

File: critiquebrainz/frontend/views/search.py [hunk 8]
```diff
@@ -18,17 +19,50 @@ def search_wrapper(query, type, offset=None):
...
     return count, results
 
 
 @search_bp.route('/')
 def index():
     query = request.args.get('query')
     type = request.args.get('type')
-    count, results = search_wrapper(query, type)
-    return render_template('search/index.html', query=query, type=type, results=results, count=count, limit=RESULTS_LIMIT)
+    if(request.args.get('review-only')=="on"):
```
Comment (gentlecat) [inline critiquebrainz/frontend/views/search.py]: Parentheses are not needed.

**Comment by gentlecat** [comment]
I can't get it to work for some reason. I pass `review-only=on` but still see all results.

**Comment by gentlecat** [comment]
Thanks for those fixes! Since we now have an ability to connect to the MusicBrainz database directly, I'd rather use that method. Would definitely be much faster and efficient.

**Pull request closed without merging**